### ISSUE TITLE
Custom Dialogs does not work with Router

### ISSUE THREAD

**Issue by tsonevn** [NativeScript/nativescript-angular#459]
_From @ignaciolarranaga on September 19, 2016 11:17_

### Did you verify this is a real problem by searching [Stack Overflow]
Yes

### Tell us about the problem
If the application has navigation (I think after you navigate) the custom dialogs stop working because this exception:

`file:///app/tns_modules/@angular/core/bundles/core.umd.js:3462:32: EXCEPTION: Uncaught (in promise): Error: Parent page is not part of the window hierarchy. Close the current modal page before showing another one!`

### Which platform(s) does your issue occur on?
Just tested on iOS

### Please provide the following version numbers that your issue occurs with:
- CLI: 2.3.0
- Cross-platform modules: 2.3.0
- Runtime(s): tns-ios 2.3.0

### Please tell us how to recreate the issue in as much detail as possible.
Check out this repo for reproduction: https://github.com/ignaciolarranaga/modal-dialog-error

Basically I:
- Created an app from scratch (tns create modal-dialog-error --ng)
- Added the custom dialog to verify it works: https://github.com/ignaciolarranaga/modal-dialog-error/commit/6b9cf2602d360dc4afe708e26ea6964561025dd4
- Added navigation to verify it starts filing: https://github.com/ignaciolarranaga/modal-dialog-error/commit/64c78b420c801ff91e1e4aa3a709e4e8f32897d3

_Copied from original issue: NativeScript/NativeScript#2759_

**Comment by tsonevn** [comment]
Hi @ignaciolarranaga,
Thank you for reporting this issue.

I reviewed your case and tested it in a sample project, however I was unable to reproduce this behavior, while using Router and Custom dialog. For your convenience I am attaching link to my [sample project](https://github.com/tsonevn/NGModalView_question38835363/blob/master/app/navigation.component.ts).

I hope this will help.
Regards,
@tsonevn

**Comment by ignaciolarranaga** [comment]
Hi @tsonevn, did you try the project I attach in the issue ?
https://github.com/ignaciolarranaga/modal-dialog-error

**Comment by tsonevn** [comment]
Hi @ignaciolarranaga,

I reviewed your project and found that the problem is due  to the missing `ModalDialogService` provider in `app.component.ts` file in your project. If you add this provider in `AppComponent`,  `ModalDialogs` will work as expected.

`app/app.component.ts`

```
import {Component} from "@angular/core";
import { ModalDialogService, ModalDialogOptions } from "nativescript-angular/modal-dialog";
import { ActivityIndicatorDialog } from "./activity-indicator-dialog.component";

@Component({
    providers: [ModalDialogService],
    selector: "my-app",
    templateUrl: "app.component.html",
})
export class AppComponent {
    public counter: number = 16;

    constructor(private _modalDialogService: ModalDialogService) {}

    
    public onTap() {
        let options: ModalDialogOptions = {
            context: { waitingText: "Waiting!!" },
            fullscreen: false
        };
        this._modalDialogService.showModal(ActivityIndicatorDialog, options);
    }
}

```


I hope this helps.
